### ISSUE TITLE
add action feature in push notification for both Android and IOS

### ISSUE THREAD

**Issue by ericowhadi** [aws-amplify/amplify-js#13011]
### Is this related to a new or existing framework?

React Native

### Is this related to a new or existing API?

Push Notifications

### Is this related to another service?

Pinpoint

### Describe the feature you'd like to request

Push notification without action support is very limited. Modern apps all have action support in notification. Without it amplify notification is lagging behind to the point I need to remove it.

### Describe the solution you'd like

actions support in push notification, in both amplify notification and pinpoint

### Describe alternatives you've considered

remove amplify notification and implement manually using other react native package implementing push notification

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by cwomack** [comment]
Hello, @ericowhadi and thanks for opening this issue.  Just to clarify the feature you're requesting a little better, is the "action support" you're referencing something along the lines of allowing users who receive a notification to have options to toggle/push once the notification is received ([this](https://developer.apple.com/documentation/usernotifications/declaring_your_actionable_notification_types) for example)?

**Comment by ericowhadi** [comment]
yes, and its equivalent in Android. Without actions, notifications are 5 years old, now all modern app using notification implement actions, and not having support for it in amplify makes the framework somewhat outdated. I am currently investigating using react-native-push-notification instead of aws-amplify/push-notification as a workaround to support actions. But I would love to stay 100% amplify, as having half-hacked solution makes our project very complex.
